gadget/gadget.go

```diff
@@ -206,6 +208,107 @@ type VolumeUpdate struct {
 	Preserve []string       `yaml:"preserve"`
 }
 
+// DiskVolumeDeviceTraits is a set of traits about a disk that were measured at
+// a previous point in time on the same device, and is used primarily to try and
+// map a volume in the gadget.yaml to a physical device on the system after the
+// initial installation is done. We don't have a steadfast and predictable way
+// to always find the device again, so we need to do a search, trying to find a
+// device which matches each trait in turn, and verify it matches the  physical
+// structure layout and if not move on to using the next trait.
+type DiskVolumeDeviceTraits struct {
+	// each member here is presented in descending order of certainty about the
+	// likelihood of being compatible if a candidate physical device matches the
+	// member. I.e. CachedDevicePath is more trusted than CachedKernelPath is
+	// more trusted than DiskID is more trusted than using the MappedStructures
+
+	// CachedDevicePath is the device path in sysfs and in /dev/disk/by-path the
+	// volume was measured and observed at during UC20+ install mode.
+	CachedDevicePath string `json:"cached-device-path"`
+
+	// CachedKernelPath is the device path like /dev/vda the volume was
+	// measured and observed at during UC20+ install mode.
+	CachedKernelPath string `json:"cached-kernel-path"`
+
+	// DiskID is the disk's identifier, it is a UUID for GPT disks or an
+	// unsigned integer for DOS disks encoded as a string in hexadecimal as in
+	// "0x1212e868".
+	DiskID string `json:"disk-id"`
+
+	// Structure contains trait information about each individual structure in
+	// the volume that may be useful in identifying whethere a disk matches a
+	// volume or not.
+	Structure []DiskStructureDeviceTraits `json:"mapped-structures"`
+}
+
+// DiskStructureDeviceTraits is a similar to DiskVolumeDeviceTraits, but is a
```

Comment:
Mmm... you call it "structure" here, but everywhere else when you document the field names it's clear that you are referring to a partition. Do we have other types of structures that are representable by this type, other tha partitions?

(if so, it would be nice to make an example in the comment)

Reply:
Thanks for the explanation! Maybe adding a line like

"A structure can be a partition or just an identifiable raw slice of the disk, such as the MBR"

would avoid this question in the future :-)

Reply:
added an explanation like this